osutil/env.go

```diff
@@ -209,6 +210,38 @@ func NewExpandableEnv(pairs ...string) ExpandableEnv {
 	return ExpandableEnv{OrderedMap: strutil.NewOrderedMap(pairs...)}
 }
 
+var envExpandRegExp, _ = regexp.Compile("[a-zA-Z0-9_]+:[+-]")
+
+// Adds support for bash conditional syntax ${VARIABLE:+XXX} and ${VARIABLE:-XXX}
```

Comment:
needs an update to the docs on the forum, does it work with snapcraft already?

Also IIRC this does not work "${FOO:-${BAR}}` and will need to be documented

Reply:
It needs an update in the docs, yes, because currently, putting that on an environment variable replaces it with an empty string. But I was waiting until a review for that.

Also, you are right, it won't work with `${FOO:-${BAR}}`. I can do a full-fledged parser, but I think that it would be excessive, and that just by documenting that limitation should be enough.